### PR TITLE
docs: add enriched events logic documentation to README.md

### PR DESCRIPTION
**Pull request by Lespoir**
## Description

Add some docs explaining how the new enriched events logic works.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -343,6 +346,56 @@ You can find your dayjs locale here:
...
+## Enriched Events Logic (Performance Boost)
+
+### Performant Data Structure
+
+We create and cache a dictionary keyed by date with all the events properties plus the overlapping counting:
+
+```
+{
+ [YYYY-MM-DD]: {
+    ...all event information,
```
Comment (nicoraynaud) [inline README.md:358]: Should this be an array of events instead ?

Comment (Lespoir) [reply README.md]: Good catch, is an array now

**Comment by Lespoir** [comment]
@acro5piano could u give it a look please

File: README.md [hunk 2]
```diff
@@ -343,6 +346,59 @@ You can find your dayjs locale here:
...
+### Multi-days events
+
+When an event spans multiple days, we create a new event for each day it spans. For example, an event that starts on 2021-05-01 and ends on 2021-05-03 will be represented as 3 events, one for each day with the following time props:
+
+```
+--- First day
+2021-05-01: {
+   ...other data
+   start: event start datetime
+   end: midnight
```
Comment (juribee) [inline README.md:376]:
```suggestion
   end: first day midnight
```

File: README.md [hunk 3]
```diff
@@ -343,6 +346,59 @@ You can find your dayjs locale here:
...
+}
+-- All the intermediate days:
+2021-05-02: {
+   ...other data
+   start: midnight
```
Comment (juribee) [inline README.md:381]:
```suggestion
   start: second day midnight
```

**Review by anton-patrushev — APPROVED** [review]

**Comment by acro5piano** [comment]
@Lespoir Thanks! Fantastic!
@juribee @anton-patrushev @nicoraynaud Thank you for your improvements!

**Pull request merged**